src/pages/RFOX/components/Unstake/UnstakeConfirm.tsx

```diff
@@ -53,14 +117,169 @@ export const UnstakeConfirm: React.FC<UnstakeRouteProps> = () => {
+    // Yeah this is arbitrary but come on, Arb is cheap
+    refetchInterval: 15_000,
+  })
+
+  const serializedUnstakeTxIndex = useMemo(() => {
+    if (!(unstakeTxid && stakingAssetAccountAddress && confirmedQuote.stakingAssetAccountId))
+      return ''
+    return serializeTxIndex(
+      confirmedQuote.stakingAssetAccountId,
+      unstakeTxid,
+      stakingAssetAccountAddress,
+    )
+  }, [confirmedQuote.stakingAssetAccountId, stakingAssetAccountAddress, unstakeTxid])
+
+  const {
+    mutateAsync: sendUnstakeTx,
+    isPending: isUnstakeMutationPending,
+    isSuccess: isUnstakeMutationSuccess,
+  } = useMutation({
+    mutationFn: async () => {
+      if (!wallet || stakingAssetAccountNumber === undefined || !stakingAsset || !callData) return
+
+      const adapter = assertGetEvmChainAdapter(stakingAsset.chainId)
+
+      const buildCustomTxInput = await createBuildCustomTxInput({
+        accountNumber: stakingAssetAccountNumber,
+        adapter,
+        data: callData,
+        value: '0',
+        to: RFOX_PROXY_CONTRACT_ADDRESS,
+        wallet,
+      })
+
+      const txId = await buildAndBroadcast({
+        adapter,
+        buildCustomTxInput,
+        receiverAddress: CONTRACT_INTERACTION, // no receiver for this contract call
+      })
+
+      return txId
+    },
+    onSuccess: (txId: string | undefined) => {
+      if (!txId) return
+
+      setUnstakeTxid(txId)
+    },
+  })
+
+  const {
+    data: userStakingBalanceOfCryptoBaseUnit,
+    isSuccess: isUserStakingBalanceOfCryptoBaseUnitSuccess,
+    queryKey: userStakingBalanceOfCryptoBaseUnitQueryKey,
+  } = useReadContract({
+    abi: foxStakingV1Abi,
+    address: RFOX_PROXY_CONTRACT_ADDRESS,
+    functionName: 'stakingInfo',
+    args: [getAddress(stakingAssetAccountAddress)], // actually defined, see enabled below
+    chainId: arbitrum.id,
+    query: {
+      enabled: Boolean(stakingAssetAccountAddress),
+      select: ([stakingBalance]) => stakingBalance.toString(),
+    },
+  })
+
+  const {
+    data: newContractBalanceOfCryptoBaseUnit,
+    isSuccess: isNewContractBalanceOfCryptoBaseUnitSuccess,
+    queryKey: newContractBalanceOfCryptoBaseUnitQueryKey,
+  } = useReadContract({
+    abi: erc20ABI,
+    address: getAddress(fromAssetId(confirmedQuote.stakingAssetId).assetReference),
+    functionName: 'balanceOf',
+    args: [getAddress(RFOX_PROXY_CONTRACT_ADDRESS)],
+    chainId: arbitrum.id,
+    query: {
+      select: data => data.toString(),
+    },
+  })
+
+  const newShareOfPoolPercentage = useMemo(
+    () =>
+      bnOrZero(userStakingBalanceOfCryptoBaseUnit)
+        .minus(confirmedQuote.unstakingAmountCryptoBaseUnit)
+        .div(newContractBalanceOfCryptoBaseUnit ?? 0)
+        .toFixed(4),
+    [
+      confirmedQuote.unstakingAmountCryptoBaseUnit,
+      newContractBalanceOfCryptoBaseUnit,
+      userStakingBalanceOfCryptoBaseUnit,
+    ],
+  )
+
+  const handleSubmit = useCallback(async () => {
+    await sendUnstakeTx(undefined)
```

Comment:
Is `undefined` a necerrary arg to this mutation function?

Reply:
Can actually be omitted altogether! 638ef1efe6
- also in the stake PR: https://github.com/shapeshift/web/pull/6925/commits/1e92047b444d5c44dea5260dbf1745ab4e238fec
- and in the change address one: https://github.com/shapeshift/web/pull/6963/commits/0766b2ffbb37087f36d310ce0616aa92287d1f58